src/engine/renderer/tr_main.cpp

```diff
@@ -1301,40 +1301,48 @@ static bool R_GetPortalOrientations( drawSurf_t *drawSurf, orientation_t *surfac
 	// locate the portal entity closest to this plane.
 	// origin will be the origin of the portal, origin2 will be
 	// the origin of the camera
-	for ( i = 0; i < tr.refdef.numEntities; i++ )
-	{
-		e = &tr.refdef.entities[ i ];
-
-		if ( e->e.reType != refEntityType_t::RT_PORTALSURFACE )
-		{
-			continue;
-		}
-
-		d = DotProduct( e->e.origin, originalPlane.normal ) - originalPlane.dist;
-
-		if ( d > 64 || d < -64 )
-		{
+	int numVertsOld = tess.numVertexes;
+	rb_surfaceTable[Util::ordinal( *( drawSurf->surface ) )]( drawSurf->surface );
+	int numVerts = tess.numVertexes - numVertsOld;
+	vec3_t portalCenter;
+	for ( int vertIndex = 0; vertIndex < numVerts; vertIndex++ ) {
+		VectorAdd( portalCenter, tess.verts[vertIndex].xyz, portalCenter );
+	}
+	VectorScale( portalCenter, 1.0 / numVerts, portalCenter );
+	float minDistance = FLT_MAX;
+	trRefEntity_t* currentPortal = nullptr;
+	for ( i = 0; i < tr.refdef.numEntities; i++ ) {
+		e = &tr.refdef.entities[i];
+
+		if ( e->e.reType != refEntityType_t::RT_PORTALSURFACE ) {
 			continue;
 		}
 
 		// get the pvsOrigin from the entity
 		VectorCopy( e->e.oldorigin, pvsOrigin );
 
 		// if the entity is just a mirror, don't use as a camera point
-		if ( e->e.oldorigin[ 0 ] == e->e.origin[ 0 ] && e->e.oldorigin[ 1 ] == e->e.origin[ 1 ] && e->e.oldorigin[ 2 ] == e->e.origin[ 2 ] )
-		{
+		if ( e->e.oldorigin[0] == e->e.origin[0] && e->e.oldorigin[1] == e->e.origin[1] && e->e.oldorigin[2] == e->e.origin[2] ) {
```

Comment:
It would be preferable to avoid whitespace changes on otherwise unmodified lines

Reply:
Fixed.